staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -144,8 +151,12 @@ type priorityInfo struct {
 
 func NewResourceManager(path string) ResourceManager {
 	scheme := runtime.NewScheme()
-	codecs := serializer.NewCodecFactory(scheme)
+	// Register conversion for apidiscovery
+	apidiscoveryv2.SchemeBuilder.Register(apidiscoveryv2conversion.RegisterConversions)
```

Comment:
is this registering into a global scheme? we just saw a data race in a unit test:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/123611/pull-kubernetes-unit/1763696339129995264

Reply:
```
==================
WARNING: DATA RACE
Read at 0x000004fe46d0 by goroutine 635:
  k8s.io/apimachinery/pkg/runtime.(*SchemeBuilder).Register()
      /home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/runtime/scheme_builder.go:39 +0xae
  k8s.io/apiserver/pkg/endpoints/discovery/aggregated.NewResourceManager()
      /home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go:155 +0x44
  k8s.io/apiserver/pkg/server.completedConfig.New()
      /home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config.go:830 +0x1be4
```

Reply:
opened https://github.com/kubernetes/kubernetes/issues/123632 to track

This is mutating the package-level schemebuilder. I think it should be adding the conversions to the local scheme instead:

```
	utilruntime.Must(apidiscoveryv2.AddToScheme(scheme))
	utilruntime.Must(apidiscoveryv2beta1.AddToScheme(scheme))
	// Add conversion for apidiscovery
	apidiscoveryv2conversion.RegisterConversions(scheme)
```